### DISCUSSION TITLE
Legendendarstellung mit PNG Dateien in GeoServer WMS Services

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#13] [Q&A]
In WebGIS4 konnten benutzerdefinierte Legenden eingebaut werden, welche im GeoServer Stil-Editor definiert wurden (
![stil_editor](https://user-images.githubusercontent.com/100088983/203489698-58c6da4d-de26-4aad-8e7f-3e389c8c3839.png)

Diese Einträge wurden dann in folgender Form dargestellt:
![legende_beispiel](https://user-images.githubusercontent.com/100088983/203489931-7b1f34db-4e9a-492f-9621-5419e3d0186e.png)

In WebGIS5 bleibt der Legendeneintrag leer:
![legende_beispiel2](https://user-images.githubusercontent.com/100088983/203491020-6a0fd8fc-dfae-4d39-81e8-ba9f28d378df.png)

Wie kann man das erreichen?

**Comment by gis-eni** [comment]
Ok, wenn ich das richtig verstehe hast du einen WMS Dienst, von dem die Legende im WebGIS5 nicht angezeigt wird. 
Normalerweise sollte der Legendenaufruf unabhängig davon sein, wie du sie erstellst. Ich kenne den GeoServer Stil Editor nicht und ob das einen Einfluss auf die *GetLegend* Methode hat.
Im WebGIS4 wurde diesbezüglich meines Wissens auch nichts spezielles dafür umgesetzt.

Wenn der Dienst im WebGIS 4 oder einen anderen Viewer (GQIS) eine Legende liefert, ist irgendwas faul... Dann wäre für mich natürlich ein Zugriff auf den entsprechenden Dienst interessant.

**Comment by gis-eni** [comment]
Das Problem ist hier, dass die GetLegendSystem Request nicht nur das Bild liefert sondern auch der Text mit Bild steht. Daher konnte kein ordentliche Legende erstellt werden. Wurde fürs erste mit einem "Hack" behoben. Schöner wäre,  wenn der Dienst wie alle anderen WMS Dienste reagieren würde.
